griptape/drivers/prompt/text_gen_driver.py

```diff
@@ -0,0 +1,43 @@
+import copy
+import requests
+from attr import define, field
+
+from griptape.artifacts import TextArtifact
+from griptape.drivers import BasePromptDriver
+
+
+@define
+class TextGenPromptDriver(BasePromptDriver):
+    """
+    Prompt Driver for the text-generation-webui https://github.com/oobabooga/text-generation-webui
+
+    params usage example:
+    https://github.com/oobabooga/text-generation-webui/blob/main/api-examples/api-example.py
+    """
+    preset: str = field(default=None, kw_only=True)
+    params: dict = field(factory=dict, kw_only=True)
+
+    model_url: str = field(default="http://localhost:5000", kw_only=True)
+    generate_uri: str = field(default="/api/v1/generate", kw_only=True)
+
+    model = field(default=None, kw_only=True)
+    tokenizer = field(default=None, kw_only=True)
+
+    def try_run(self, value: str) -> TextArtifact:
+
+        url = f"{self.model_url}{self.generate_uri}"
+
+        if self.preset is None:
+            request = copy.deepcopy(self.params)
```

Comment:
Why do we need to copy `params`?

Reply:
the copy is used because the params used as a request body with addition of the prompt. dont want to add the prompt to the params field. 
the copy result is the body of the request + prompt.